### DISCUSSION TITLE
WLAN Association timeout and frequency band

### DISCUSSION THREAD

**Discussion by u77345** [rsta2/circle#421] [Development]
Never getting a clear association for the first attempt. I wonder if the association and DHCP timeout can be extended, and the frequency defaulted to the 5GHz band? 

Below is a log for the 'normal' scenario. There are more retries in ~20% of the cases.

Circle 45.3.1 started on Raspberry Pi 3 Model A+ (AArch64)
00:00:00.66 timer: SpeedFactor is 1.51
00:00:01.80 dwhci: No device connected to root port
00:00:01.83 sdhost: emmc1: sdhost-bcm2835 loaded
00:00:02.34 emmc: Found a valid version 2.00 SD card
00:00:02.48 wlan: ether4330: chip 0x4345 rev 6 type 1
00:00:03.08 wlan: ether4330: firmware ready
00:00:03.09 wlan: ether4330: addr B8:27:EB:8D:XX:XX
00:00:03.12 kernel: Compile time: Dec 10 2023 21:37:22
00:00:03.24 wpa: Setting country code to 'US'
00:00:06.65 wpa: Trying to associate with 74:ac:b9:e7:xx:xx (SSID='xxxxx' freq=2462 MHz)
00:00:07.25 dhcp: No response from server. Retrying.
00:00:09.70 wpa: Associated with 74:ac:b9:e7:xx:xx
00:00:10.54 wpa: WPA: Key negotiation completed with 74:ac:b9:e7:xx:xx [PTK=CCMP GTK=CCMP]
00:00:10.55 wpa: CTRL-EVENT-CONNECTED - Connection to 74:ac:b9:e7:xx:xx completed (auth) [id=0 id_str=]
00:00:15.25 dhcp: No response from server. Retrying.
00:00:19.33 dhcp: No response from server. Retrying.
00:00:19.33 dhcp: IP address is 10.0.1.12

**Comment by rsta2** [comment]
The process of establishing a WLAN connection takes some time. The DHCP 
process is not synchronized with the WLAN. It is started, when the network 
subsystem is started and retries after 2, 4, 8, 16, 32 ... seconds. These 
timeouts are hard-coded. Normally the WLAN connection should be up after some 
seconds and DHCP should succeed on the next retry.

One cannot use the frequency selection parameters in wpa_supplicant.conf, 
because the WLAN driver does not support it. But you can select the WLAN AP 
using the ssid= and bssid= settings in this file and give the AP a priority=, 
if you have more than one AP.

**Reply by u77345** [reply]
Thanks Rene!
